visualization/include/visualization/circle.hpp

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "point.hpp"
+
+namespace visualization::geometry {
+
+    class Circle {
+
+        public:
+            Circle(const Point& center, double radius);
+
+            double get_radius() const;
+            Point get_center() const;
+
+        private:
+            const Point center;
```

Comment:
should not be const because you are blocking the copy assignement default implementation (and the move as well)